### PR TITLE
Refactor data store

### PR DESCRIPTION
**Pull request by aintgoin2goa**
This is a complete re-write of the data layer.  Partly cos my initial implementation didn't make any sense.

Data is now separated out into buckets, eacb with it's own poller and validation etc.

This made it easy to split the graphql queries into chunks, which seems to make it faster and will make it easy to poll for topstories often but other stuff less often.

Should also make it easier in the future to try different architectures (ie doing the polling on workers instead of web nodes)

### CODE REVIEW

**Comment by leggsimon** [comment]
Just my comments about splitting up the front page query into smaller components, also maybe it would be good to gitignore those really massive graphql responses?

File: server/data/sources/topstories.datasource.js [hunk 1]
```diff
@@ -4,6 +4,6 @@ const GraphQLDataSource = require('./graphql.datasource');
 module.exports = class GraphQLTopStoriesDataSource extends GraphQLDataSource {
 
 	constructor ({region = ''}){
-		super({query, variables:{region:region}});
+		super({query, variables:{region:region}, interval:30000});
```
Comment (leggsimon) [inline server/data/sources/topstories.datasource.js]: for consistency with `server/data/sources/otherhomepage.datasource.js`
```js
constructor ({region = ''}){
    const variables = {region};
    super({query, variables});
}
```

File: server/sections/video/presenter.js [hunk 2]
```diff
@@ -5,8 +5,8 @@ const path = require('path');
 
 const view = presenter.loadView(path.resolve(__dirname, 'view.html'));
 
-module.exports = function videoSectionPresenter ({flags}){
-	const data = dataStore.retrieve('otherHomePage');
+module.exports = function videoSectionPresenter ({flags, edition}){
+	const data = dataStore.retrieve(`${edition}OtherHomePage`);
```
Comment (leggsimon) [inline server/sections/video/presenter.js]: ditto

File: server/sections/technology/presenter.js [hunk 3]
```diff
@@ -9,8 +9,8 @@ const views = {
 }
 
 
-module.exports = function technologyPresenter ({flags}){
-	const data = dataStore.retrieve('otherHomePage');
+module.exports = function technologyPresenter ({flags, edition}){
+	const data = dataStore.retrieve(`${edition}OtherHomePage`);
```
Comment (leggsimon) [inline server/sections/technology/presenter.js]: ditto

File: server/sections/premium/presenter.js [hunk 4]
```diff
@@ -5,8 +5,8 @@ const path = require('path');
 
 const view = presenter.loadView(path.resolve(__dirname, 'view.html'));
 
-module.exports = function premiumPresenter ({flags}){
-	const data = dataStore.retrieve('otherHomePage');
+module.exports = function premiumPresenter ({flags, edition}){
+	const data = dataStore.retrieve(`${edition}OtherHomePage`);
```
Comment (leggsimon) [inline server/sections/premium/presenter.js]: `edition` should probably be `region`? (in fact in all the presenters)

File: server/sections/regional-news/presenter.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ module.exports = function regionalNewsPresenter ({flags, region, edition}){
 	const data = dataStore.retrieve('regionalNews');
 	const topStoriesData = dataStore.retrieve(`${edition}TopStories`);
```
Comment (leggsimon) [inline server/sections/regional-news/presenter.js]: ditto

File: server/sections/regional-news/presenter.js [hunk 6]
```diff
@@ -24,7 +24,7 @@ module.exports = function regionalNewsPresenter ({flags, region, edition}){
...
 	const usElectionData = dataStore.retrieve('usElection');
-	const opinionData = dataStore.retrieve('otherHomePage').opinion;
+	const opinionData = dataStore.retrieve(`${edition}OtherHomePage`).opinion;
```
Comment (leggsimon) [inline server/sections/regional-news/presenter.js]: ditto

File: server/sections/premium/presenter.js (hunk 4 above)
Comment (aintgoin2goa) [inline server/sections/premium/presenter.js]: Ideally graphql should be updated to use Edition not region - and to not use "US", which is misleading as it's not us at all.

**Comment by ironsidevsquincy** [comment]
We'll sort out Edition/Region-gate in the new api (and I guess the old one too while we're maintaining both)

**Comment by callumlocke** [comment]
this will probably conflict with my PR (#1144), but not too badly, so go ahead and merge this when you're happy with it and I'll rebase.

**Pull request merged**